protocol/avdl/keybase1/incoming-share.avdl

```diff
@@ -24,4 +24,11 @@ protocol incomingShare {
   }
 
   array<IncomingShareItem> getIncomingShareItems();
+
+  enum IncomingShareCompressPreference {
+    ORIGINAL_0,
+    COMPRESSED_1
```

Comment:
can this be a struct that also has the chat/kbfs preference instead?

Reply:
It always goes to chat now, and there’s a footer button allowing you to go to kbfs. I can’t think of another setting right now, but if we do in the future perhaps we can use a different key with the same type in the DB key?

Reply:
we'd have to migrate old entries, i think in general it's better to have a struct so we can add fields at our leisure. \

Reply:
hmm not sure we have to migrate in that case. Currently the key for this compress preference is:

```
libkb.DbKey{
	Typ: libkb.DBIncomingSharePreference,
	Key: "v0-compress-preference",
}
```

If we need more, we can just use 

```
libkb.DbKey{
	Typ: libkb.DBIncomingSharePreference,
	Key: "v0-other-preference",
}
```

I like that this makes them independent from each other. Other wise when we set a value we have to read the old struct out first, change the field we want to change, then write it back. Either that, or deal with the struct in TS too.